appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/ContextSetupProviderImpl.java

```diff
@@ -206,12 +207,21 @@ public void reset(ContextHandle contextHandle) {
     }
 
     private boolean isApplicationEnabled(String appId) {
+        boolean result = false;
         if (appId != null) {
             Application app = applications.getApplication(appId);
-            if (app != null)
-                return deployment.isAppEnabled(app);
+            if (app != null) {
+                result = deployment.isAppEnabled(app);
+            } else { 
+                // if app is null then it is likely that appId is still deploying
+                // and its enabled status has not been written to the domain.xml yet
+                // this can happen for example with a Startup EJB submitting something
+                // it its startup method. Reference Payara GitHub issue 204              
+                logger.info("Job submitted for " + appId + " likely during deployment. Continuing...");
+                result = true;
```

Comment:
so you are making a bet that every unknown `appId` is just a "deployment-in-progress" and it'll be enabled eventually? is it a good bet?  what if the `appId` is genuinely unknown? what if it had been just undeployed?